angular/src/app/shared/components/home/table/table.component.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  Component,
+  ChangeDetectionStrategy,
+  ViewChild,
+  AfterViewInit,
+  ElementRef,
+} from '@angular/core';
+
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+
+import { debounceTime, distinctUntilChanged, fromEvent, map, merge, Observable, startWith, switchMap, tap } from 'rxjs';
+
+import { TableFilms, TableFilmsParameters } from '../../../../core/services/table-films';
+import { Film } from '../../../../core/models/film';
+import { FilmDto } from '../../../../core/mappers/dto/film.dto';
+
+/**
+ * Table component.
+ */
+@Component({
+  selector: 'sw-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TableComponent implements AfterViewInit {
+
+  public constructor(private readonly database: TableFilms) {
+  }
+
+  /** Columns to show. */
+  public readonly displayedColumns: readonly string[] = ['title', 'director', 'producer', 'created'];
+
+  /** Length of films for table with chosen parameters.*/
+  public resultsLength = 0;
+
+  /** Flag is data loading.*/
+  public isLoadingResults = true;
+
+  /** Flag is rate limit is reached.*/
+  public isRateLimitReached = false;
+
+  /** Elements on page.*/
+  public pageSize = 2;
+
+  /** Index of previous page.*/
+  public previousPage!: number;
+
+  /** Films for table. */
+  public films$!: Observable<Film[]>;
+
+  @ViewChild(MatPaginator, { static: false })
+
+  /**
+   * @inheritDoc
+   * Reference on Paginator.
+   */
+  private paginator!: MatPaginator;
+
+  @ViewChild(MatSort)
+
+  /**
+   * @inheritDoc
+   * Reference on sort table.
+   */
+  private sort!: MatSort;
+
+  /**
+   * @inheritDoc
+   * Reference on input, field.
+   */
+  @ViewChild('input') private input!: ElementRef<HTMLInputElement>;
```

Comment:
Place such decorators and their field on separate lines:
```suggestion
  @ViewChild('input')
  private input!: ElementRef<HTMLInputElement>;
```